app/presenters/tutorial_view_step_presenter.rb

```diff
@@ -0,0 +1,52 @@
+require 'tutorial_view_chapter_presenter'
+
+class StepPresenter
+  def initialize(step, chapter)
+    @step = step
+    @step_order = nil
+    @chapter = chapter
+  end
+
+  def step_ordering()
+    ordering = {}
+    counter = 0
+    @chapter.steps.each do |step|
+      if step.warning?
+        ordering[step] = "WARNING"
```

Comment:
Refers to ordering more than self (maybe move it to another class?) (FeatureEnvy)